cpp/include/cuvs/neighbors/hnsw.hpp

```diff
@@ -51,6 +56,31 @@ struct index_params : cuvs::neighbors::index_params {
       is parallelized with the help of CPU threads.
    */
   int num_threads = 0;
+
+  /** HNSW M parameter: number of bi-directional links per node (used when building with ACE).
+   */
+  size_t m = 32;
+
+  /** Parameters to fine tune GPU graph building. By default we select the parameters based on
+   * dataset shape and HNSW build parameters. You can override these parameters to fine tune the
+   * graph building process as described in the CAGRA build docs.
+   *
+   * Set ace_params to configure ACE (Augmented Core Extraction) parameters for building
+   * a GPU-accelerated HNSW index. ACE enables building indexes for datasets too large
+   * to fit in GPU memory.
+   *
+   * @code{.cpp}
+   * hnsw::index_params params;
+   * // Configure ACE parameters
+   * params.graph_build_params = hnsw::graph_build_params::ace_params();
+   * auto& ace = std::get<hnsw::graph_build_params::ace_params>(params.graph_build_params);
+   * ace.npartitions = 4;
+   * ace.ef_construction = 120;
```

Comment:
We have `hnsw::index_params::ef_construction` and `ace_params::ef_construction`. We need to clarify which one to use. I recommend to use `hnsw::index_params::ef_construction`.